test/activations.jl

```diff
@@ -315,15 +283,35 @@ end
     end
 end
 
-@testset "AutoDiff" begin
+## Autodiff tests
+
+UNARY_FUNCTIONS = [@eval($a) for (a, _) in NNlib.UNARY_ACTS]
+
+BINARY_FUNCTIONS = [@eval($a) for (a, _, _) in NNlib.BINARY_ACTS]
+
+has_rule(a) = rrule(a, 1f0) === nothing ? "(no rule)" : ""
+
+@testset "Gradient inference" begin
+    @testset "$(a): $(has_rule(a))" for a in ACTIVATION_FUNCTIONS
+        @testset "$T" for T in [Float16, Float32, Float64]
+            for val in [-10, -1, 0, 1, 10]
+                grad = @inferred gradient(a, T(val))
+                @test typeof(grad[1]) == T
```

Comment:
This replaces `test_gradient_float_precision_preserving`. I thought something was broken, so I made it print more stuff. But in the end it's not broken. Just the tests are easier to read.